balance_controller:
  virtual_model_controller:
    heading:
      kp: 600
      kd: 10
      kff: 1
    lateral:
      kp: 600
      kd: 10
      kff: 1
    vertical:
      kp: 600
      kd: 10
      kff: 1
    roll:
      kp: 500
      kd: 10
      kff: 0.2
    pitch:
      kp: 500
      kd: 8
      kff: 0.2
    yaw:
      kp: 500
      kd: 5
      kff: 1
    gravity_compensation_percentage: 1.005
  contact_force_distribution:
    weights:
      force:
        heading: 1
        lateral: 1
        vertical: 1
      torque:
        roll: 10
        pitch: 10
        yaw: 5
      regularizer:
        value: 0.00001
      force_diff:
        value: 1
        is_use: false
    constraints:
      friction_coefficient: 0.6
      minimal_normal_force: 10
      load_factor: 1
single_leg_controller:
  x_direction:
    kp: 150
    kd: 5
  y_direction:
    kp: 150
    kd: 5
  z_direction:
    kp: 150
    kd: 5

mpc:
  weights: [5, 5, 0.2, 0, 0, 10, 0.1, 0.1, 1., 1., 1., 1., 0]
  #weights: [0.25, 0.25, 0.2, 2, 2, 50, 0, 0, 0.3, 0.2, 0.2, 0.1,0]
  #weights: [0.25, 2, 10, 50, 50, 100, 0, 0, 0.3, 0.2, 0.2, 0.1, 0.0]
  #roll-pitch-yaw x-y-z ang-vel vel gravity-place-holder
  #weights: [20, 20, 50.0, 10, 10, 100, 10.0, 10.0, 1.0, 1.0, 1.0, 1.0, 0]
  roll: 0.25
  pitch: 0.25
  yaw: 10
  x: 2
  y: 2
  z: 20
  roll_vel: 0.1
  pitch_vel: 0.1
  yaw_vel: 0.3
  x_vel: 0.2
  y_vel: 0.2
  z_vel: 0.2
  gravity_holder: 0.0
  steps_MPC: 8
  #mi:10
  delta_t_MPC: 0.05
  #mi:0.026
  torque_weight: 0.00001
  #mi:4e-5
  body_mass: 20.73
